File: app/src/main/java/com/qwert2603/spenddemo/model/ChangeKind.java
package com.qwert2603.spenddemo.model;

import android.support.annotation.IntDef;

@IntDef(value = {ChangeKind.INSERT, ChangeKind.UPDATE, ChangeKind.DELETE})
public @interface ChangeKind {
    int INSERT = 0;
    int UPDATE = 1;
    int DELETE = 2;
}

File: spend/src/main/java/com/qwert2603/spenddemo/model/entity/ChangeKind.kt
package com.qwert2603.spenddemo.model.entity

import com.qwert2603.syncprocessor.entity.ChangeKind as SyncChangeKind

enum class ChangeKind {
    INSERT, UPDATE, DELETE
}

fun ChangeKind.toSyncChangeKind() = when(this) {
    ChangeKind.INSERT -> SyncChangeKind.CREATE
    ChangeKind.UPDATE -> SyncChangeKind.EDIT
    ChangeKind.DELETE -> SyncChangeKind.DELETE
}

fun SyncChangeKind.toChangeKind() = when(this) {
    SyncChangeKind.CREATE -> ChangeKind.INSERT
    SyncChangeKind.EDIT -> ChangeKind.UPDATE
    SyncChangeKind.DELETE -> ChangeKind.DELETE
}